docs/api/getting_started_with_oauth.md

```diff
@@ -0,0 +1,206 @@
+---
+title: OAuth2
+---
+
+# Getting started with OAuth2
+
+OpenChatStudio uses OAuth2 with the Authorization Code Flow with PKCE (Proof Key for Code Exchange) to enable secure third-party integrations.
+
+## How OAuth 2.0 works
+
+For a detailed explanation of the OAuth 2.0 authorization code flow, see [OAuth 2 Simplified](https://aaronparecki.com/oauth-2-simplified/#authorization). OpenChatStudio follows this standard flow, with specific endpoints documented below.
+
+## OpenChatStudio OAuth Endpoints
+
+| Endpoint | URL |
+|----------|-----|
+| Authorization | `https://www.openchatstudio.com/o/authorize/` |
+| Token | `https://www.openchatstudio.com/o/token/` |
+
+## Step 1: Register your application with OpenChatStudio
+
+Contact the OpenChatStudio team to register your application.
+
+You'll receive:
+
+- **Client ID**: A public identifier for your application
+- **Client secret**: A confidential secret used for token exchange (keep this secure and server-side only!)
+
+## Step 2: Initiate the authorization call
+
+Your application should redirect the user to OpenChatStudio's authorization endpoint to request permission.
+
+### PKCE Setup (Required)
+
+For security, you must implement PKCE:
+
+1. Generate a random `code_verifier` (43-128 characters, unreserved characters)
+2. Create a `code_challenge` by SHA256 hashing the verifier and base64url encoding it
+3. Include the `code_challenge` in your authorization request
+
+Here's example Python code to generate PKCE parameters:
+
+```python
+import secrets
+import string
+import base64
+import hashlib
+
+# Generate a random code verifier (43-128 characters)
+characters = string.ascii_letters + string.digits + '-._~'
+code_verifier = ''.join(secrets.choice(characters) for _ in range(128))
+
+# Create the code challenge by hashing and encoding the verifier
+code_challenge = base64.urlsafe_b64encode(
+    hashlib.sha256(code_verifier.encode()).digest()
+).decode().rstrip('=')
+```
+
+### Query Parameters
+
+| Parameter | Required | Description |
+|-----------|----------|-------------|
+| `response_type` | Yes | Must be `code` |
+| `client_id` | Yes | Your client ID |
+| `redirect_uri` | Yes | The URL where you want to receive the authorization code. Must match a registered URI for your application |
+| `code_challenge` | Yes | The PKCE code challenge (base64url-encoded SHA256 hash of your code_verifier) |
+| `code_challenge_method` | Yes | Must be `S256` (SHA256) |
+| `state` | Recommended | Random string to prevent CSRF attacks. Store this and validate the response |
+| `scope` | No | Space-separated list of scopes. See available scopes in the [API docs](https://openchatstudio.com/api/docs/). If omitted, defaults to all scopes |
+| `team` | No | Specific team to scope the token to |
```

Comment:
@SmittieC it's worth noting what happens if no team is provided. I also realized that even if a team is provided, the user could still change it. What do you think about disabling that? ie. just display the team if it's provided in the URL otherwise allow the user to choose.

Reply:
No objections from my side. I'll make a quick update. With that update, this piece can stay as-is

Reply:
https://github.com/dimagi/open-chat-studio/pull/2509